src/App.js

```diff
@@ -36,9 +38,63 @@ class App extends Component{
   }
 
   navigateHome = () => {
-    this.setState({isSelected: false, singleMovie: {}, error: ''})
+    this.setState({filteredMovies: ''})
   }
 
+  // sortMovies = (category) => {
+  //   if (category === 'newToOldRelease') {
+  //     let filtered = this.state.movies.sort((movA, movB) => new Date(movA.release_date) - new Date(movB.release_date))
+  //     this.setState({ filteredMovies: { ...filtered } })
+  //   } else if (category === 'oldToNewRelease') {
+  //     let filtered = this.state.movies.sort((movA, movB) => new Date(movB.release_date) - new Date(movA.release_date))
+  //     this.setState({ filteredMovies: { ...filtered } })
+  //   } else if (category === 'ratings') {
+  //     let filtered = this.state.movies.sort((movA, movB) => (movA.average_rating) - (movB.average_rating))
+  //     this.setState({ filteredMovies: { ...filtered } })
+  //   }
+  // }
+
+  filterByTitle = ({textInput, dateInput, ratingInput}) => {
+    console.log('firing')
+    console.log('FBT', ratingInput)
+    let text = textInput.toLowerCase();
+    let filtered;
+    if (textInput.length) { 
+      filtered = this.state.movies.filter(movie => movie.title.toLowerCase().includes(text))
+      this.setState({ filteredMovies: [ ...filtered ] })
+    } else if (dateInput.length > 0) {
+      filtered = this.state.movies.filter(movie => {
+        let compareDate = movie.release_date.split('-').splice(0, 2).join('-');
+        if (compareDate === dateInput) {
+          return movie
+        } 
+      })
+      this.setState({ filteredMovies: [ ...filtered ] })
+    } else if (ratingInput.length) {
+      filtered = this.state.movies.filter(movie => {
+        let rating = movie.average_rating / 2
+        if (rating >= ratingInput) {
+          return movie
+        }
+      })
+      this.setState({ filteredMovies: [...filtered] })
+    } else {
+      console.log('Sorry, there are no matches')
+    }
+    console.log(this.state.filteredMovies)
+  }
+
+  // filterMovies = (input) => {
+  //   let filtered = this.state.movies.filter(movie => {
+  //     if (input === movie.average_rating.toString() || movie.release_date) {
+  //       this.setState({ filteredMovies: { ...filtered } })
+  //     } else {
+  //       this.setState({filteredMovies: 'Sorry, there are no matches'})
+  //     }
+  //   })
+  // }
+  
+
   render() {
```

Comment:
I am also curious about figuring out how to move our logic from the render method into the individual components?